defaults/main.yml

```diff
@@ -55,9 +55,9 @@ datadog_additional_groups: {}
 datadog_agent_major_version: ""
 
 # Pin agent to a version. Highly recommended.
-# Defaults to the latest version of the major version chosen in datadog_agent_major_version
-# If both datadog_agent_major_version and datadog_agent_version are set, they must be
-# compatible (ie. the major version in datadog_agent_version must be datadog_agent_major_version)
+# Defaults to the latest version of the major version chosen in agent_datadog_agent_major_version
```

Comment:
There's probably a reason behind, but you have twice `agent` in the same name

Reply:
All variable must be prefixed with ``agent_`` because of the ``ansible-lint`` expected role variable format in collections. Since a large number of variable were replaced, It was done automatically hence why it replaced comments variable as well.